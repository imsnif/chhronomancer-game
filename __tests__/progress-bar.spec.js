import { Text } from 'react-native'
import React from 'react'
import ProgressBar from '../components/progress-bar'
import renderer from 'react-test-renderer'

jest.mock('mobx-react/native', () => require('mobx-react/custom'))

test('ProgressBar (default props) => renders correctly', () => {
  const tree = renderer.create(
    <ProgressBar />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('ProgressBar (with progress) => renders correctly', () => {
  const tree = renderer.create(
    <ProgressBar progress={50}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('ProgressBar (with children) => renders correctly', () => {
  const tree = renderer.create(
    <ProgressBar>
      <Text>foo</Text>
    </ProgressBar>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('ProgressBar (with progress and children) => renders correctly', () => {
  const tree = renderer.create(
    <ProgressBar progress={50}>
      <Text>foo</Text>
    </ProgressBar>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
