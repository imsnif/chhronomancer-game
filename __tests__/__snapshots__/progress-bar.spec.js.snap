// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar (default props) => renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#00ff00",
      "fontFamily": "telegrama_raw",
      "fontSize": 7,
      "marginLeft": 9,
      "marginTop": 15,
    }
  }
>
  Questing [ =======------------------------------------------- ] 20% 00:01:23
</Text>
`;

exports[`ProgressBar (with children) => renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#00ff00",
      "fontFamily": "telegrama_raw",
      "fontSize": 7,
      "marginLeft": 9,
      "marginTop": 15,
    }
  }
>
  Questing [ =======------------------------------------------- ] 20% 00:01:23
</Text>
`;

exports[`ProgressBar (with progress and children) => renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#00ff00",
      "fontFamily": "telegrama_raw",
      "fontSize": 7,
      "marginLeft": 9,
      "marginTop": 15,
    }
  }
>
  Questing [ =======------------------------------------------- ] 20% 00:01:23
</Text>
`;

exports[`ProgressBar (with progress) => renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#00ff00",
      "fontFamily": "telegrama_raw",
      "fontSize": 7,
      "marginLeft": 9,
      "marginTop": 15,
    }
  }
>
  Questing [ =======------------------------------------------- ] 20% 00:01:23
</Text>
`;
